mdpow/workflows/base.py

```diff
@@ -0,0 +1,166 @@
+# MDPOW: base.py
+# 2022 Cade Duckworth
+
+"""
+:mod:`mdpow.workflows.base` --- Automated workflow base functions
+=================================================================
+
+.. autofunction:: directory_paths
+.. autofunction:: directory_iteration
+
+"""
+
+import os
+import re
+import pandas as pd
+
+from mdpow.workflows import registry
+
+import logging
+
+logger = logging.getLogger('mdpow.workflows.base')
+
+def directory_paths(parent_directory=None, csv=None, csv_save_dir=None):
+    """Takes a top directory containing MDPOW projects and determines
+       the molname, resname, and path, of each MDPOW project within.
+
+       Optionally takes a .csv file containing `molname`, `resname`, and
+       `paths`, in that order. 
+
+       :keywords:
+
+       *parent_directory*
+           the path for the location of the top directory 
+           under which the subdirectories of MDPOW simulation
+           data exist, additionally creates a 'dir_paths.csv' file
+           for user manipulation of metadata and for future reference
+
+       *csv*
+           .csv file containing the molecule names, resnames,
+           and paths, in that order, for the MDPOW simulation
+           data to be iterated over must contain header of the
+           form: `molecule,resname,path`
+
+       *csv_save_dir*
+           optionally provided directory to save .csv file, otherwise,
+           data will be saved in current working directory
+
+       :returns:
+
+       *directory_paths*
+           :class:`pandas.DataFrame` containing MDPOW project metadata
+
+       .. rubric:: Example
+       
+       Typical Workflow::
+
+           directory_paths = directory_paths(parent_directory='/foo/bar/MDPOW_projects')
+           directory_iteration(directory_paths)
+
+           or
+
+           directory_paths = directory_paths(csv='/foo/bar/MDPOW.csv')
+           directory_iteration(directory_paths)
```

Comment:
do proper code formatting

Reply:
Also, can you look at the `.. autodata:: registry` portion of registry.py and let me know if there is a way to suppress the type hint. If you look at the docs build it included instructions on how to build a dictionary underneath the dictionary values for registry.